pkg/store/bucket.go

```diff
@@ -722,12 +721,16 @@ func blockSeries(
 			s.refs = append(s.refs, meta.Ref)
 		}
 		if len(s.chks) > 0 {
+			if err := chunksLimiter.Reserve(uint64(len(s.chks))); err != nil {
```

Comment:
We could potentially limit chunk bytes instead TBH, but this is good for a start

Reply:
Or both. In Cortex we would need by count :)

Reply:
Opened an issue: https://github.com/thanos-io/thanos/issues/2861